packages/collector/src/agentConnection.js

```diff
@@ -486,3 +493,26 @@ function logLargeSpans(spans) {
     )}`
   );
 }
+
+/**
+ *
+ * @param {Array.<InstanaBaseSpan>} spans
+ */
+function getSpanLengthInfo(spans) {
+  const spanMapping = {
+    1: 'entrySpans',
+    2: 'exitSpans'
+  };
+
+  const countBySpanType = spans.reduce((acc, item) => {
+    // @ts-ignore
```

Comment:
Either we can ignore line by line or for file itself

Reply:
Why do we need to ignore it in the first place?

Reply:
Otherwise, there is a type error warning in that line.

item is of type `InstanaBaseSpan` and in there `k` is an optional attribute. So there is a type error

Reply:
Also in the very next line, acc is an instance of InstanaBaseSpan in the context of array.reduce function. So it is also throwing a lint warning

Reply:
`ts-ignore` should only be used if there is no other solution.
I think there are solutions because its our own types?

Reply:
Added type definitions in the file for type safety. Thus removing the `ts-ignore`